crates/metaslang/bindings/src/lib.rs

```diff
@@ -312,10 +369,11 @@ impl<'a, KT: KindTypes + 'static> Definition<'a, KT> {
             .and_then(|info| info.definiens.clone())
     }
 
-    pub fn get_file(&self) -> Option<&'a str> {
+    pub fn get_file(&self) -> FileDescriptor {
         self.owner.stack_graph[self.handle]
             .file()
-            .map(|file| self.owner.stack_graph[file].name())
+            .and_then(|file| FileDescriptor::from_string(self.owner.stack_graph[file].name()).ok())
+            .unwrap_or_else(|| unreachable!("Definition does not have a valid file descriptor"))
```

Comment:
nit: Is `from_string()` ever expected to fail here? if not, I suggest propagating the error instead of silencing it here:

```suggestion
            .and_then(|file| FileDescriptor::from_string(self.owner.stack_graph[file].name()))
            .expect("Definition does not have a valid file descriptor")
```

Reply:
`from_string()` should not fail here since we're constructing the `Definition/Reference` with valid nodes. The `ok()` is there to appease the type checker by making the result from `from_string()` and `file()` compatible. Hmm... Changing the return type of `from_string()` to `Option<FileDescriptor>` would make this clearer. Plus, `FileDescriptorError` was a marker for error, but it doesn't carry any other information and it's not used anywhere explicitly.

Reply:
I think `Result<>` here would be more idiomatic, as it would `Error` if the input was invalid. `Option<>` would be used in cases where valid input can actually exist without having a conversion. So I think given that you want to prevent panicking in `from_string()`, keeping your `Result<>` would do exactly that. This is similar to Rust's APIs like `usize::from_string(str) -> Result<>`/

But up to you. I don't think it is worth spending more of your time if you see otherwise.